### PR TITLE
Add alternative component model using a neural network

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

File: bugbug/model.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def __init__(self, lemmatization=False):
         self.undersampling_enabled = True
         self.cross_validation_enabled = True
 
+        self.calculate_importance = True
+        self.calculate_classification_metrics = True
```
Comment (marco-c) [inline bugbug/model.py]: This one is important to compare models, so we should try to make it work

File: bugbug/model.py [hunk 2]
```diff
@@ -52,6 +55,9 @@ def get_important_features(self, cutoff, shap_values):
 
         return top_features
 
+    def calculate_labels(self, classes, bugs):
+        return np.array([classes[bug['id']] for bug in bugs()])
```
Comment (marco-c) [inline bugbug/model.py]: We could make each model directly return the labels in the format they want in `get_labels`

File: bugbug/models/triage.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+            'short_desc_vocab_size': self.short_desc_vocab_size,
+            'short_desc_emb_sz': self.short_desc_emb_sz,
+            'long_desc_maxlen': self.long_desc_maxlen,
+            'long_desc_vocab_size': self.long_desc_vocab_size,
+            'long_desc_emb_sz': self.long_desc_emb_sz
+        }
+        self.clf = TriageClassifier(**kwargs)
+
+    def get_labels(self):
+        products = {
```
Comment (marco-c) [inline bugbug/models/triage.py]: We can reuse the get_labels from ComponentModel

File: bugbug/models/triage.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
```
Comment (marco-c) [inline bugbug/models/triage.py]: Given that this is implementing the same as the components model but with a different ML algorithm, maybe it's better to call the file something like `component_nn.py` (nn as in neural networks).

File: run.py [hunk 5]
```diff
@@ -19,13 +19,19 @@
...
+                        help='Goal of the classifier',
+                        choices=['bug', 'regression', 'tracking', 'qaneeded', 'uplift', 'component', 'devdocneeded', 'triage'],
+                        default='bug')
     parser.add_argument('--classify', help='Perform evaluation', action='store_true')
     parser.add_argument('--generate-sheet', help='Perform evaluation on bugs from last week and generate a csv file', action='store_true')
     args = parser.parse_args()
 
     model_file_name = f'{args.goal}model'
 
+    probabilities = True
```
Comment (marco-c) [inline run.py]: Nit: let's call these `probabilities_enabled` and `importances_enabled`, so you don't have to rename the pre-existing `importances` variable

File: bugbug/model.py (hunk 2 above)
Comment (johngian) [reply bugbug/model.py]: Can we tackle this in a different PR, since it involved changes to alla the models we currently have.

Comment (johngian) [reply bugbug/model.py]: Can we tackle this in a different PR since it involves changes in all the rest models as well ?

Comment (marco-c) [reply bugbug/model.py]: OK!

**Comment by johngian** [comment]
@marco-c PR updated

**Comment by marco-c** [comment]
@johngian I've made a couple refactorings, could you double check this is still working for you?

**Pull request merged**